common/script/script-api/src/main/java/org/thingsboard/script/api/tbel/TbDate.java

```diff
@@ -155,8 +230,214 @@ private static DateTimeFormatOptions getDateFormattingOptions(String options) {
+    }
+    public void setUTCFullYear(int year, int month, int date) {
+        parseInstant(createDateTimeFromPattern(year, month, date, getUTCHours(), getUTCMinutes(), getUTCSeconds(), getUTCMilliseconds()));
+    }
+    public void setUTCMonth(int month) {
+        parseInstant(createDateTimeFromPattern(getUTCFullYear(), month, getUTCDate(), getUTCHours(), getUTCMinutes(), getUTCSeconds(), getUTCMilliseconds()));
+    }
+    public void setUTCMonth(int month, int date) {
+        parseInstant(createDateTimeFromPattern(getUTCFullYear(), month, date, getUTCHours(), getUTCMinutes(), getUTCSeconds(), getUTCMilliseconds()));
+    }
+    public void setUTCDate(int date) {
+        parseInstant(createDateTimeFromPattern(getUTCFullYear(), getUTCMonth(), date, getUTCHours(), getUTCMinutes(), getUTCSeconds(), getUTCMilliseconds()));
+    }
+    public void setUTCHours(int hrs) {
+         parseInstant(createDateTimeFromPattern(getUTCFullYear(), getUTCMonth(), getUTCDate(), hrs, getUTCMinutes(), getUTCSeconds(), getUTCMilliseconds()));
+    }
+    public void setUTCHours(int hrs, int minutes) {
+         parseInstant(createDateTimeFromPattern(getUTCFullYear(), getUTCMonth(), getUTCDate(), hrs, minutes, getUTCSeconds(), getUTCMilliseconds()));
+    }
+    public void setUTCHours(int hrs, int minutes, int seconds) {
+         parseInstant(createDateTimeFromPattern(getUTCFullYear(), getUTCMonth(), getUTCDate(), hrs, minutes, seconds, getUTCMilliseconds()));
+    }
+    public void setUTCHours(int hrs, int minutes, int seconds, int ms) {
+         parseInstant(createDateTimeFromPattern(getUTCFullYear(), getUTCMonth(), getUTCDate(), hrs, minutes, seconds, ms));
+    }
+    public void setUTCMinutes(int minutes) {
+       parseInstant(createDateTimeFromPattern(getUTCFullYear(), getUTCMonth(), getUTCDate(), getUTCHours(), minutes, getUTCSeconds(), getUTCMilliseconds()));
+    }
+    public void setUTCMinutes(int minutes, int seconds) {
+       parseInstant(createDateTimeFromPattern(getUTCFullYear(), getUTCMonth(), getUTCDate(), getUTCHours(), minutes, seconds, getUTCMilliseconds()));
+    }
+    public void setUTCMinutes(int minutes, int seconds, int ms) {
+       parseInstant(createDateTimeFromPattern(getUTCFullYear(), getUTCMonth(), getUTCDate(), getUTCHours(), minutes, seconds, ms));
+    }
+    public void setUTCSeconds(int seconds) {
+        parseInstant(createDateTimeFromPattern(getUTCFullYear(), getUTCMonth(), getUTCDate(), getUTCHours(), getUTCMinutes(), seconds, getUTCMilliseconds()));
+    }
+    public void setUTCSeconds(int seconds, int ms) {
+        parseInstant(createDateTimeFromPattern(getUTCFullYear(), getUTCMonth(), getUTCDate(), getUTCHours(), getUTCMinutes(), seconds, ms));
+    }
+    public void setUTCMilliseconds(int ms) {
+        parseInstant(createDateTimeFromPattern(getUTCFullYear(), getUTCMonth(), getUTCDate(), getUTCHours(), getUTCMinutes(), getUTCSeconds(), ms));
```

Comment:
You are converting instant to String and then back to Instant. I'm afraid that's not right. 

You should use methods of ZonedDateTime or similar. See https://docs.oracle.com/javase/8/docs/api/java/time/ZonedDateTime.html#withMonth-int-